src/crdt.rs

```diff
@@ -870,7 +870,7 @@ impl Crdt {
         blob_recycler: &BlobRecycler,
     ) -> Option<SharedBlob> {
         let pos = (ix as usize) % window.read().unwrap().len();
-        if let Some(blob) = &window.read().unwrap()[pos].data {
+        if let Some(ref mut blob) = &mut window.write().unwrap()[pos].data {
```

Comment:
this write() makes the window more of a Mutex<> than a RwLock<>

the as_mut() stuff is better than the RwLock<> stuff that was there before?